### PR TITLE
Replace flake8 with ruff

### LINKED ISSUE #12610: light_minimal theme is missing
It accidentally duplicates the caliber theme due to a copy-pasta. 

ref: https://github.com/bokeh/bokeh/pull/12605/files#r1020946545

This was noticed in #12605 and is fixed by that PR if it is merged

### PR DESCRIPTION
**Pull request by bryevdv**
Hi @bokeh/core I became aware of [ruff](https://github.com/charliermarsh/ruff) as a possible alternative to `flake8` implemented in Rust, and created a branch just to experiment. I quickly became convinced that `ruff` is a dramatically better alternative, and I would very strongly encourage us to switch entirely. 

* Ruff is ~150-200x faster than flake8 on my machine, scanning the whole repo takes ~0.2s instead of ~20s. This is an enormous quality of life improvement for local dev. It's fast enough that I added it as an actual commit hook, which is terrific. 
* Shaving tens of seconds off several CI jobs will be nice as well
* Ruff replaces two packages `flake8` and `flake8-pyi` with a single package, and also allows top-level configuration to be further consolidated in to `pyproject.toml`
* Ruff supports many more checks that we could consider adding over time
* Ruff has finer grained exclusions, we can now properly check all the `__init__.py` we were skipping entirely, previously
* Ruff detected several problems and improvements that flake8 was not, including at least one legitimate bug (see comment inline)

It's also potentially possible that Ruff could replace `isort` at some point as well. `isort` is the other very slow codebase check, so that would be nice. But I think it's too early for that (esp since we have different configs in different subdirs)

* fixes #12610

### CODE REVIEW

File: src/bokeh/themes/__init__.py [hunk 1]
```diff
@@ -93,7 +93,7 @@
 built_in_themes = {
     CALIBER       : Theme(json=_caliber.json),
     DARK_MINIMAL  : Theme(json=_dark_minimal.json),
-    LIGHT_MINIMAL : Theme(json=_caliber.json),
+    LIGHT_MINIMAL : Theme(json=_light_minimal.json),
```
Comment (bryevdv) [inline src/bokeh/themes/__init__.py:96]: This was an actual bug, not sure how long it has been this way

File: examples/basic/data/color_mappers.py [hunk 2]
```diff
@@ -2,11 +2,11 @@
 log mapping and linear mapping with different color palette.
 
 .. bokeh-example-metadata::
-    :apis: bokeh.plotting.Figure.scatter, bokeh.models.ColumnDataSource, bokeh.models.annotations.ColorBar, bokeh.models.mappers.LinearColorMapper, bokeh.models.mappers.LogColorMapper # noqa: E501
+    :apis: bokeh.plotting.Figure.scatter, bokeh.models.ColumnDataSource, bokeh.models.annotations.ColorBar, bokeh.models.mappers.LinearColorMapper, bokeh.models.mappers.LogColorMapper
     :refs: :ref:`ug_topics_images_colormapped`
     :keywords: color, tools, scatter, data_map
 
-'''
+''' # noqa: E501
```
Comment (bryevdv) [inline examples/basic/data/color_mappers.py:9]: Some day we will get automatic api-linking in example code and do away with `:apis:` alogether, but until then, this is much nicer to work with long docstring lines.

File: examples/models/buttons.py [hunk 3]
```diff
@@ -8,11 +8,11 @@
 radio buttons, dropdown menu, checkbox and radio with groups.
 
 .. bokeh-example-metadata::
-    :apis: bokeh.models.widgets.buttons.Button, bokeh.models.widgets.buttons.Dropdown, bokeh.models.widgets.buttons.Toggle, bokeh.models.widgets.groups.CheckBoxButtonGroup, bokeh.models.widgets.groups.CheckboxGroup, bokeh.models.widgets.groups.RadioButtonGroup, bokeh.models.widgets.groups.RadioGroup  # noqa: E501
+    :apis: bokeh.models.widgets.buttons.Button, bokeh.models.widgets.buttons.Dropdown, bokeh.models.widgets.buttons.Toggle, bokeh.models.widgets.groups.CheckBoxButtonGroup, bokeh.models.widgets.groups.CheckboxGroup, bokeh.models.widgets.groups.RadioButtonGroup, bokeh.models.widgets.groups.RadioGroup
```
Comment (mattpap) [inline examples/models/buttons.py:11]: Can't this line be split up or alternatively non-qualified names resolved docs compilation time (afteral we have a flat namespace for pretty much all models)?

Comment (bryevdv) [reply examples/models/buttons.py:11]: Lots of things could be done, I haven't had time, and I was made aware that there may be an existing sphinx plugin to link all API functions in example code blocks automatically. My priority is evaluating that (when I have the time) and if that does not work, then making QOL improvements to our extension.

Comment (bryevdv) [reply examples/models/buttons.py:11]: Oh but to answer the primary question: no, at the moment it cannot be split up

**Comment by bryevdv** [comment]
@bokeh/core unless there are comments/objections, I will plan to merge this Wednesday.

**Comment by bryevdv** [comment]
Absent any further input in the last two days, I will merge this now.

**Comment by bryevdv** [comment]
brand new versin of ruff caught some additional spurious `# noqa` after merge, I pushed a tiny commit to fix directly:

https://github.com/bokeh/bokeh/commit/1f73a5a0a6dfee9bceef7fd75b06ad764b4ad681

**Pull request merged**